cognitedata/reveal
# fix: Proper handling of mouse wheel zoom for non static Reveal "windows"

#### Type of change

![Bug](https://img.shields.io/badge/Type-Bug-red) 

#### Jira ticket :blue_book:


https://cognitedata.atlassian.net/browse/REV-570

## Description :pencil:

Additional fix to PR https://github.com/cognitedata/reveal/pull/2641 .

Fix for mouse wheel zooming in `zoomToCursor` and `zoomPastCursor` camera controls modes when Reveal viewer is not a static object within browser page (i.e. in the docs or any other app where viewer size or position is changing). Previous approach was using `element.offsetLeft/Top` properties for calculating position of the reference element within client viewport. This PR changes it to using `element.getBoundingClientRect()` method internally which results in correct values even when reveal `domElement` is partially invisible when scrolled.

## How has this been tested? :mag:


On modified example in the viewer with added html element that expands html page and makes it scrollable downwards. With previous implementation when reveal viewer was partially visible or just not in the "initial" position (the one when html page is loaded), zooming with mouse wheel was done to wrong part of the screen. After this PR, it is getting correct values.

## Test instructions :information_source:

Open documentation preview on any example and scroll down a little but so that the viewer is still visible. Try to zoom in on any part of the model. 
**Expected behaviour**: camera is zooming in to the point exactly under the cursor.
**Previous behaviour**: camera is zooming in to the point with some offset from the cursor. (can be checked in master documentation on version next)



## Checklist :ballot_box_with_check:
